### DISCUSSION TITLE
利用doris构建小规模实时数仓咨询

### DISCUSSION THREAD

**Discussion by wandayou** [apache/doris#23232] [A - General / Q&A]
刚刚涉及构建小规模实时数仓，向其他团队应用 提供业务数据的各种维度统计，设计思路如下：
1.doris中增加ods层unique模型，保存业务数据的变更需求(flink cdc实现)
2.增加dwd层，进行表数据维度退化，将ods业务表进行各种join后变大宽表，方便查询及聚合
3.增加dws层，保存对dwd层数据的汇总结果，然后提供应用查询

请问这种规划方式合理吗？另外如何从ods层到dwd层数据生成，以及从dwd到dws层的数据生成，有什么好且简单的实现方式吗？

**Comment by GaoYang20** [comment]
我们这边和你的方案差不多，第二部用的flink sql实现的宽表，部分使用的Rollup，小数据量目前没什么问题
